compiler/ballerina-lang/src/main/java/io/ballerina/projects/Settings.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.projects;
+
+import io.ballerina.projects.internal.DefaultDiagnosticResult;
+import io.ballerina.projects.internal.model.Central;
+import io.ballerina.projects.internal.model.Proxy;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+
+import java.util.Collections;
+
+/**
+ * Represents a Settings.toml file.
+ *
+ * @since 2.0.0
+ */
+public class Settings {
+    private final Proxy proxy;
+    private final Central central;
+    private final DiagnosticResult diagnostics;
+
+    private Settings(Proxy proxy, Central central, DiagnosticResult diagnostics) {
```

Comment:
Since we are not accumulating any diagnostics here, IMO we can keep diagnostics in SettingsBuilder without passing it down here. During settings creation, this would allow to check for diagnostics from the SettingsBuilder and create the Settings instance if there are no errors.